sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -247,6 +251,125 @@ def test_lasso_cv_positive_constraint():
+    # in the pipeline and with normalize set to False
+
+    n_samples, n_features = 100, 2
+    random_state = np.random.RandomState(0)
+    w = random_state.randn(n_features)
+    X = random_state.randn(n_samples, n_features)
+    X += 20  # make features non-zero mean
+
+    y = X.dot(w)
+    # make classes out of regression
+    if 'Classifier' in str(test_model):
+        y[y > np.mean(y)] = -1
+        y[y > 0] = 1
+    if 'MultiTask' in str(test_model):
+        y = np.stack((y, y), axis=1)
+
+    X, X_test, y, y_test = train_test_split(X, y, random_state=42)
+
+    # normalize is True
+    clf_norm = test_model(normalize=True, fit_intercept=True,
+                          **args)
+    clf_norm.fit(X, y)
+    y_pred_norm = clf_norm.predict(X_test)
+
+    if 'alpha' in args:
+        alpha_scaled = args['alpha']
+    if 'Lasso' in str(test_model) and test_model != LassoLarsIC:
+        alpha_scaled = alpha_scaled * np.sqrt(X.shape[0])
+    if 'Ridge' in str(test_model) and 'alpha' in args:
+        alpha_scaled = alpha_scaled * X.shape[0]
+    if 'Elastic' in str(test_model):
+        if args['l1_ratio'] == 1:
+            alpha_scaled = alpha_scaled * np.sqrt(X.shape[0])
+        if args['l1_ratio'] == 0:
+            alpha_scaled = alpha_scaled * X.shape[0]
+
+    if 'alpha' in args:
+        args['alpha'] = alpha_scaled
+
+    clf_pipe = make_pipeline(
+        StandardScaler(),
+        test_model(normalize=False, fit_intercept=True, **args)
+    )
+    clf_pipe.fit(X, y)
+    y_pred_pipe = clf_pipe.predict(X_test)
+
+    assert_array_almost_equal(clf_norm.coef_ * clf_pipe[0].scale_,
+                              clf_pipe[1].coef_, decimal=5)
+    assert clf_pipe[1].intercept_ == pytest.approx(y.mean())
+    assert (clf_norm.intercept_ ==
+            pytest.approx(y.mean() - clf_norm.coef_.dot(X.mean(0))))
+    assert_array_almost_equal(y_pred_norm, y_pred_pipe)
+
+
+@pytest.mark.parametrize(
+    "test_model, args",
+    [
+     (Lasso, {"tol": 1e-16, "alpha": 0.1}),
+     (LassoCV, {"tol": 1e-16}), (ElasticNetCV, {}),
+     (RidgeClassifier, {"solver": 'sparse_cg', "alpha": 0.1}),
+     (ElasticNet, {"tol": 1e-16, 'l1_ratio': 1, "alpha": 0.1}),
+     (ElasticNet, {"tol": 1e-16, 'l1_ratio': 0, "alpha": 0.1}),
+     (Ridge, {"solver": 'sparse_cg', 'tol': 1e-12, "alpha": 0.1}),
+     (LinearRegression, {}),
+     (RidgeCV, {})
+     ])
+def test_model_pipeline_same_dense_and_sparse(test_model, args):
+    # Test that linear model preceeded by StandardScaler in the pipeline and
+    # with normalize set to False gives the same y_pred and the same .coef_
+    # given X sparse or dense
+
+    rng = np.random.RandomState(0)
+    n_samples = 200
+    n_features = 2
+    X = rng.randn(n_samples, n_features)
+    X[X < 0.1] = 0.
+
+    X_sparse = sparse.csr_matrix(X)
+    y = rng.rand(n_samples)
+
+    if 'Classifier' in str(test_model):
+        y = np.sign(y)
+
+    clf_pipe_dense = make_pipeline(
```

Comment:
Let's create the pipeline at the begining.
Do not use `clf` but `model` -> `model_dense`/`model_sparse`.